terraform/eks-cloudwatch/main.tf

```diff
@@ -0,0 +1,158 @@
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# -------------------------------------------------------------------------
+
+# creating a eks cluster takes around 10 minutes typically.
+# so in the eks/k8s test, we need tester to provide the cluster instead of creating it in terraform
+# so that we can shorten the execution time
+
+variable "aws_account_id" {
+  default = ""
+}
+
+variable "sample_app_image_repo" {
+  default = "611364707713.dkr.ecr.us-west-2.amazonaws.com/otel-test/container-insight-samples"
+}
+
+module "common" {
+  source = "../common"
+
+  aoc_image_repo = var.aoc_image_repo
+  aoc_version = var.aoc_version
+}
+
+# region
+provider "aws" {
+  region = var.region
+}
+
+# get eks cluster by name
+data "aws_eks_cluster" "testing_cluster" {
+  name = var.eks_cluster_name
+}
+
+data "aws_eks_cluster_auth" "testing_cluster" {
+  name = var.eks_cluster_name
+}
+
+# set up kubectl
+provider "kubernetes" {
+  config_path = "~/.kube/config"
+  version = "~> 1.13"
+}
+
+# create a unique namespace for each run
+resource "kubernetes_namespace" "aoc_ns" {
+  metadata {
+    name = "aoc-ns-${module.common.testing_id}"
+  }
+}
+
+resource "kubernetes_service_account" "aoc_role" {
+  metadata {
+    name = "aoc-role-${module.common.testing_id}"
+    namespace = kubernetes_namespace.aoc_ns.metadata[0].name
+  }
+  automount_service_account_token = true
+}
+
+resource "kubernetes_cluster_role_binding" "aoc_role_binding" {
+  metadata {
+    name = "aoc-role-binding-${module.common.testing_id}"
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind = "ClusterRole"
+    name = "cluster-admin"
+  }
+  subject {
+    kind = "ServiceAccount"
+    name = "aoc-role-${module.common.testing_id}"
+    namespace = kubernetes_namespace.aoc_ns.metadata[0].name
+  }
+}
+resource "kubernetes_deployment" "standalone_aoc_deployment" {
+  metadata {
+    name = "aoc"
+    namespace = kubernetes_namespace.aoc_ns.metadata[0].name
+    labels = {
+      app = "aoc"
+    }
+  }
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        app = "aoc"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "aoc"
+        }
+      }
+
+      spec {
+        service_account_name = "aoc-role-${module.common.testing_id}"
+        automount_service_account_token = true
+        # aoc
+        container {
+          name = "aoc"
+          image = module.common.aoc_image
+          image_pull_policy = "Always"
+          args = [
+            "--config",
+            "/etc/eks/prometheus/config-all.yaml"]
```

Comment:
i might missed this part, who generates this config?

Reply:
Sorry about the confusion. This integration test will be introduced in the PR: https://github.com/aws-observability/aws-otel-collector/pull/394

Reply:
Q: should we use `config-all.yaml`? Or make it as variable and each test case (eg, appmesh, nginx. etc) only apply their own configurations?

Reply:
In the final PR of this integration test series, we test all the workload all at once. Metrics and logs are separated by namespaces, testing workload one by one makes no difference in validation with testing in a batch. However, testing separately surely will make integration test more expensive because of the extra effort on building and compiling and initiating new containers for testing. Also test order should be taken care of, which is what terraform is poor at. 

CloudWatch agent doesn't even have separated config for each workload. Frankly, I'm still questioning myself whether those separated configs are valuable.

Reply:
i'm not sure if that's a good approach to bake all the testing configurations into image and release it to customer? for example, if we want to test with negative endpoint where we need to modify the endpoint value in config, how would we do that?

Reply:
my point is not about we should do it right away, but instead, we need to think about what's the overlapping parts we are making with the eks folder, how do we extract the common parts, if there's a contributor who wants to add a new testcase, how do we guide him to add.

Reply:
Not sure if I understand your concern correctly, but if `the testing configurations` refer to the [prometheus container insight configuration](https://github.com/aws-observability/aws-otel-collector/blob/main/config/eks/prometheus/config-all.yaml), it's not TESTing configurations, but official configurations that can be used by the end users when they'd like to use aoc for container insight hands on. Same as what [cloudwatch agent prometheus-eks.yaml](https://github.com/aws/amazon-cloudwatch-agent/blob/master/amazon-cloudwatch-container-insights/k8s-deployment-manifest-templates/deployment-mode/service/cwagent-prometheus/prometheus-eks.yaml) is doing.

And I still remember the conversation we had on why not merge into eks foler: https://github.com/aws-observability/aws-otel-test-framework/pull/247#discussion_r600981387.

I, as a contributor, am confused what eks folder is served for. Maybe a document to illustrate the existing eks module would be greatly helpful for me to follow.